### PR TITLE
Add implementation for #[blanket(derive(Arc))] with associated tests

### PR DESCRIPTION
**Pull request by najamelan**
This implements derive(Arc).

I hope I didn't miss anything, but it works on my crate and all tests are passing.

NOTE: rebased on #3. You probably want to merge that first.

### CODE REVIEW

File: src/derive/arc.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+pub fn derive(trait_: &syn::ItemTrait) -> syn::Result<syn::ItemImpl> {
+    // build the methods
+    let mut methods: Vec<syn::ImplItemMethod> = Vec::new();
+    for item in trait_.items.iter() {
+        if let syn::TraitItem::Method(ref m) = item {
+            if let Some(receiver) = m.sig.receiver() {
+                match receiver {
+                    syn::FnArg::Receiver(r) if r.mutability.is_some() => {
+                        let msg = "cannot derive `Rc` for a trait declaring `&mut self` methods";
```
Comment (althonos) [inline src/derive/arc.rs]: Please update the message for `Arc` instead of `Rc` here !

File: tests/derive_arc/fails/receiver_box.stderr [hunk 2]
```diff
@@ -0,0 +1,5 @@
+error: cannot derive `Rc` for a trait declaring methods with arbitrary receiver types
```
Comment (althonos) [inline tests/derive_arc/fails/receiver_box.stderr]: This should read *cannot derive `Arc` for a trait...* here as well.

**Review by althonos — CHANGES_REQUESTED** [review]
Overall looks good, and it should work as `Rc` so I don't see any thing else to add/change. Please just update the error messages (that are still for `Rc` instead of `Arc`) and this should be clear for merging!

File: src/derive/arc.rs (hunk 1 above)
Comment (najamelan) [reply src/derive/arc.rs]: ah, missed those. Will do. I will rebase the other PR's.

File: tests/derive_arc/fails/receiver_box.stderr (hunk 2 above)
Comment (najamelan) [reply tests/derive_arc/fails/receiver_box.stderr]: This one is strange, as the tests pass...

Comment (najamelan) [reply tests/derive_arc/fails/receiver_box.stderr]: Oh, I see it's the message from the derive above.

**Comment by najamelan** [comment]
Hopefully this is good now.

**Comment by althonos** [comment]
Just run `cargo fmt` and we'll be good to go :wink:

**Comment by najamelan** [comment]
Ok, I've inserted a cargo fmt. I didn't realize before that I had a cargo config file in a parent directory. That's why it looked like it was going to change the whole crate.

**Comment by althonos** [comment]
No problem, thanks !

**Pull request merged**